From 45f5ff439e9034e4b4d40127f1367bebf71ecfb4 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 11 Jun 2020 11:41:51 -0700
Subject: [PATCH 1377/1852] dt-binding: soc: xilinx: ai-engine: Add FPGA bridge
 for AI engine partition

commit f1a1c5e29bed444ff3679a6ac5f27fdab8fb34d3 from
https://github.com/Xilinx/linux-xlnx.git

Add FPGA bridge example as AI engine driver adds FPGA bridge support for
AI engine partition.

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/soc/xilinx/xlnx,ai-engine.yaml   | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml b/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
index 7aa69e101381..581739f8688c 100644
--- a/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
+++ b/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
@@ -53,6 +53,9 @@ patternProperties:
       engine device. Each AI engine partition is isolated from the other
       AI engine partitions. An AI engine partition is defined by Xilinx
       platform design tools.
+      AI engine partition driver will create a FPGA bridge when probes the
+      AI engine partition. The AI engine SHIM row of the partition is the
+      bridge to connect AI engine partition, FPGA and PS.
 
     properties:
       reg:
@@ -87,3 +90,20 @@ examples:
         xlnx,partition-id = <1>;
       };
     };
+
+  - |
+    /*
+     * Device tree overlay example to for FPGA partial configuration which
+     * connects to AI engine partition.
+     */
+    fpga_region0: fpga_region0 {
+      fpga-bridges = <&aie_partition0>;
+    };
+    fragment@0 {
+      target = <&fpga_region0>;
+      #address-cells = <1>;
+      #size-cells = <1>;
+      __overlay__ {
+        firmware-name = "partition.pdi";
+      };
+    };
-- 
2.31.1

